packages/legend-query-builder/src/components/QueryBuilder.tsx

```diff
@@ -444,6 +503,12 @@ export const QueryBuilder = observer(
                             Show Watermark
                           </MenuContentItemLabel>
                         </MenuContentItem>
+                        <MenuContentItem onClick={openCheckEntitlmentsEditor}>
+                          <MenuContentItemIcon>{null}</MenuContentItemIcon>
```

Comment:
You don't have to pas null here if you don't want to add an icon. Just remove `MenuContentItemIcon`